### ISSUE TITLE
Text formating button: insertion of a link

### ISSUE THREAD

**Issue by desnoes** [c2corg/v6_ui#1576]
I made some test in this document: http://www.demov6.camptocamp.org/outings/828446/en/tete-de-vauclave-toto-fait-du-ski

- [ ]  The insertion of a link does not work.

- [ ]  ~~(There is a problem when using ordered and unordered lists successively, only the first type of list is selected. If there is another balise in between, it works fine.)~~

**Comment by asaunier** [comment]
> There is a problem when using ordered and unordered lists successively, only the first type of list is selected.

Does this use case really happen?
Who need to have ordered and unordered lists successively?

**Comment by asaunier** [comment]
> The insertion of a link does not work.

Could you precise what does not work?

I have just tried to use the link button and indeed observed a strange result:
* the button create some code similar to 
```
[enter link description here](http://example.com)
```
(which is a regular mardown link notation)
* but then when previewing the document or saving it a link is created with the following URL:
```
http://www.demov6.camptocamp.org/outings/edit/828446/a%20href=%22http://example.com%22%3Ehttp://example.com%3C/a
```
or
```
http://www.demov6.camptocamp.org/outings/828446/en/a%20href=%22http://example.com%22%3Ehttp://example.com%3C/a
```

=> I suspect there is a conflict between the bbcode/markdown parsers. It seems the first one replaces the raw URL by ``<a href="http://example.com">http://example.com</a>`` which creates the following code
```
[enter link description here](<a href="http://example.com">http://example.com</a>)
```
Then the markdown parser converts this syntax to 
```
<a href="<a href="http://example.com">http://example.com</a>">enter link description here</a>
```
=> the link is converted twice!

We might have to remove the auto converting of URLs to links.
What about disabling the bbcode parser once for all?

**Comment by desnoes** [comment]
> Does this use case really happen?
Who need to have ordered and unordered lists successively?

I agree this use case is very unlikely to happen. But the bug is still there.

> Could you precise what does not work?

When i click on the link, i have the message 'Page non trouvée'.

> What about disabling the bbcode parser once for all?

What is the impact on legacy documents ?

**Comment by asaunier** [comment]
> I agree this use case is very unlikely to happen. But the bug is still there.

But what's the point of spending time on something "very unlikely to happen"?

In addition this behaviour is not related to the work done for the project, it's specific to the (external) markdown parser used to interpret the markdown code. So if this is really an issue, it should rather be reported to https://github.com/waylan/Python-Markdown/issues

> When i click on the link, i have the message 'Page non trouvée'.

Yes, as explained in my previous comment, a double conversion is done that generates a wrong URL.

> What is the impact on legacy documents ?

Disabling the bbcode parser will simply make the bbcode remain unconverted in the documents using bbcode (displayed as is).
The result will depend on the amount of bbcode used in the documents. If no bbcode is used (or very few), the document will be displayed as expected. If a lot of bbcode is used, the document will be poorly formatted with several bbcode tags appearing unconverted.

I think in old documents, the following bbcode is used:
* bold ``[b]``, italic ``[i]``, underline ``[u]``, strikethrough ``[s]``
* link ``[url=http://example.com]Example[/url]`` as well as the direct conversion of raw URL
but equivalent alternative tags exist in markdown (that are now provided by the formatting toolbar)

Other tags are either:
* standard markdown code (titles such as ``## Montée``, ordered/unordered lists, etc.)
* extended tags specific to C2C, using extensions of the markdown parser such as images, wikilinks, video, important/warning tags, "toc"
* not (yet?) supported specific tags such as ``[col]``, ``L#`` etc.

Using 2 formatting syntaxes at the same time is a no sense, at mean term I recommend to get rid of the bbcode syntax anyway. It means that the existing bbcode tags should be replaced by markdown equivalents.

**Comment by desnoes** [comment]
> Using 2 formatting syntaxes at the same time is a no sense, at mean term I recommend to get rid of the bbcode syntax anyway. It means that the existing bbcode tags should be replaced by markdown equivalents.

I agree with you, but we need to prepare a script to do the translation BBCode-Markdown automatically. It concerns many many documents (outings, routes, articles....). 

> But what's the point of spending time on something "very unlikely to happen"?

If the bug is easy to fix, it worth it even if it barely happens. I did not observe this bug in the forum.

**Comment by asaunier** [comment]
>  I did not observe this bug in the forum.

Because Discourse probably don't use the same tool to parse the markdown syntax. It's a completely distinct system and cannot be compared.

**Comment by asaunier** [comment]
> The insertion of a link does not work.

This part is most likely fixed by PR #1661
